loadtest/control/gencontroller/utils.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package gencontroller
+
+import (
+	"errors"
+	"math/rand"
+)
+
+// pickAction randomly selects an action from a map of userAction with
+// probability proportional to the action's frequency.
+func pickAction(actions map[string]userAction) (*userAction, error) {
```

Comment:
The other one takes a slice (not a map) and of a different type (`userAction` is not exported). It's something we should probably do at some point but I'd leave it to a separate PR since it's not a simple fix.